src/main/src/main/java/org/geoserver/catalog/impl/ModificationProxyCloner.java

```diff
@@ -194,7 +195,13 @@ public int compare(Class c1, Class c2) {
             // nothing to copy
             return null;
         }
-        Collection<T> copy = source.getClass().newInstance();
+        Collection<T> copy;
+        try {
+            copy = source.getClass().newInstance();
+        } catch (InstantiationException e) {
+            //we'll just pick something
+            copy = new ArrayList<T>();
```

Comment:
This something is I suppose what you needed, but what if the original collection was a Set or a Map?
What was the class that could not be instantiated?